input/psbt.go

```diff
@@ -0,0 +1,144 @@
+// of UnknownOption.
+func wrapUnknownOption(key string, value []byte) UnknownOption {
+	return func() *psbt.Unknown {
+		return &psbt.Unknown{
+			Key:   []byte(key),
+			Value: value,
+		}
+	}
+}
+
+// uint64Bytes returns a byte slice for the big endian representation of the
+// argument.
+func uint64Bytes(num uint64) []byte {
+	var msgBytes [8]byte
+
+	byteOrder.PutUint64(msgBytes[:], num)
+
+	return msgBytes[:]
+}
+
+// uint64Bytes returns a byte slice for the big endian representation of the
+// argument.
+func uint32Bytes(num uint32) []byte {
+	var msgBytes [4]byte
+
+	byteOrder.PutUint32(msgBytes[:], num)
+
+	return msgBytes[:]
+}
+
+// boolBytes returns a single-byte slice with a 0 for false and 1 for true.
+func boolBytes(val bool) []byte {
+	if val {
+		return []byte{1}
+	}
+
+	return []byte{0}
+}
+
+// descBytes returns a byte slice representing the descriptor (family/index
+// and possibly pubkey).
+func descBytes(desc *keychain.KeyDescriptor) []byte {
+	// Allocate 8 bytes for the locator, 33 for possible pubkey.
+	msgBytes := make([]byte, 8, 41)
+
+	byteOrder.PutUint32(msgBytes[:4], uint32(desc.Family))
+	byteOrder.PutUint32(msgBytes[4:], desc.Index)
+
+	if desc.PubKey != nil {
+		msgBytes = append(msgBytes, desc.PubKey.
+			SerializeCompressed()...)
+	}
+
+	return msgBytes
+}
+
+// ChannelType returns an UnknownOption for the channel type.
+func ChannelType(chanType uint64) UnknownOption {
+	return wrapUnknownOption(OptChanType, uint64Bytes(chanType))
+}
+
+// Initiator returns an UnknownOption for whether we initiated the channel.
+func Initiator(initiator bool) UnknownOption {
+	return wrapUnknownOption(OptInitiator, boolBytes(initiator))
+}
+
+// CsvDelay returns an UnknownOption for the CSV delay.
+func CsvDelay(delay uint32) UnknownOption {
+	return wrapUnknownOption(OptCsvDelay, uint32Bytes(delay))
+}
+
+// CltvExpiry returns an UnknownOption for the CLTV expiry.
+func CltvExpiry(expiry uint32) UnknownOption {
+	return wrapUnknownOption(OptCltvExpiry, uint32Bytes(expiry))
+}
+
+// LeaseExpiry returns an UnknownOption for the lease expiry.
+func LeaseExpiry(expiry uint32) UnknownOption {
+	return wrapUnknownOption(OptLeaseExpiry, uint32Bytes(expiry))
+}
+
+// CommitPoint returns an UnknownOption for the commit point.
+func CommitPoint(point *btcec.PublicKey) UnknownOption {
+	return wrapUnknownOption(OptCommitPoint, point.SerializeCompressed())
+}
+
+// RHash returns an UnknownOption for the rHash.
+func RHash(rHash []byte) UnknownOption {
+	return wrapUnknownOption(OptRHash, rHash)
+}
+
+// RemoteKey returns an UnknownOption for the remote key.
+func RemoteKey(key *btcec.PublicKey) UnknownOption {
+	return wrapUnknownOption(OptRemoteKey, key.SerializeCompressed())
+}
+
+// LocalDesc returns an UnknownOption for the local descriptor.
+func LocalDesc(desc *keychain.KeyDescriptor) UnknownOption {
+	return wrapUnknownOption(OptLocalDesc, descBytes(desc))
```

Comment:
This would require a bit more encoding/decoding effort and a bit more space, but would be more "standard" if we used the full BIP32 derivation path here instead, see https://github.com/btcsuite/btcd/blob/master/btcutil/psbt/bip32.go#L65

Reply:
I'm still working on this, as it requires passing the master key fingerprint and network parameters/coin type through to the PSBT.